nav2_route/src/goal_intent_extractor.cpp

```diff
@@ -53,6 +59,15 @@ void GoalIntentExtractor::configure(
   nav2_util::declare_parameter_if_not_declared(
     node, "min_dist_from_start", rclcpp::ParameterValue(0.10));
   min_dist_from_start_ = static_cast<float>(node->get_parameter("min_dist_from_start").as_double());
+
+  std::string global_costmap_topic;
+  nav2_util::declare_parameter_if_not_declared(
+    node, "global_costmap_topic", rclcpp::ParameterValue(std::string("global_costmap/costmap_raw")));
+  node->get_parameter("global_costmap_topic", global_costmap_topic);
+
+  costmap_sub_ = std::make_unique<nav2_costmap_2d::CostmapSubscriber>(node, global_costmap_topic);
```

Comment:
Where is the nearest costmap subscriber, can we not just pull one in from somewhere else (or are the rest in plugins?) 

If there's no centrally located one, can we make a centrally located one that can be pulled in here and also used in the plugins API?